services/ui_backend_service/data/cache/client/cache_server.py

```diff
@@ -234,26 +262,19 @@ def validate_actions(self, actions):
                                            % (mod_str, cls_str))
 
     def schedule(self):
-
         def queued_request(queue):
             while queue:
                 yield queue.popleft()
 
         for request in chain(queued_request(self.hi_prio_requests),
                              queued_request(self.lo_prio_requests)):
 
-            try:
-                worker = Worker(request, self.filestore)
-                if worker.tempdir:
-                    # NOTE: Cleanup this logic.
-                    worker.start()
-                    return worker
-            except CacheFullException:
-                echo("Cache storage is full, worker failed to start!")
-                return None
+            worker = Worker(request, self.filestore)
+            if worker.tempdir:
+                worker.start()
+                return worker
```

Comment:
will the `start()` always succeed? the `new_worker()` in `loop()` also has some optimistic assumptions about workers.

Reply:
Additional error handling now in place.